### ISSUE TITLE
Remove all __repr__s defined in beacon/types

### ISSUE THREAD

**Issue by s0b0lev** [ethereum/trinity#802]
### What is wrong?
Remove all `__repr__`s defined in beacon/types and just use the automatic defn from py-ssz

Example of issue:
```python
In [25]: head = chaindb.get_canonical_head(BeaconBlock)

In [26]: head
Out[26]: Traceback (most recent call last):
  File "/home/s0b0lev/.pyenv/versions/3.7.3/envs/trinity/lib/python3.7/site-packages/IPython/core/formatters.py", line 224, in catch_format_error
    r = method(self, *args, **kwargs)
  File "/home/s0b0lev/.pyenv/versions/3.7.3/envs/trinity/lib/python3.7/site-packages/IPython/core/formatters.py", line 702, in __call__
    printer.pretty(obj)
  File "/home/s0b0lev/.pyenv/versions/3.7.3/envs/trinity/lib/python3.7/site-packages/IPython/lib/pretty.py", line 400, in pretty
    return _repr_pprint(obj, self, cycle)
  File "/home/s0b0lev/.pyenv/versions/3.7.3/envs/trinity/lib/python3.7/site-packages/IPython/lib/pretty.py", line 695, in _repr_pprint
    output = repr(obj)
  File "/home/s0b0lev/.pyenv/versions/3.7.3/envs/trinity/lib/python3.7/site-packages/eth2/beacon/types/blocks.py", line 137, in __repr__
    f'<Block #{self.slot} '
AttributeError: 'BeaconBlock' object has no attribute 'num_attestations'
```

### How can it be fixed

- [ ] Remove all `__repr__`s defined in beacon/types and just use the automatic defn from py-ssz
- [ ] Override via a custom `__str__` instance on those types that need it

**Comment by ralexstokes** [comment]
a PR to address this issue should work off of https://github.com/ethereum/trinity/pull/791

**Comment by ChihChengLiang** [comment]
Support the idea that replacing __repr__ with __str__.

An automatic `__repr__` in `py-ssz` sounds nice too. But need to think about how to handle big objects like state that has many big scary fields.

**Comment by ralexstokes** [comment]
also noting we do get an automatic `__str__` instance in `py-ssz`.

so i would say we only need to overwrite `__str__` where necessary and prefer the default thing unless it is unmanageable for some reason like cc is saying

**Comment by ralexstokes** [comment]
Addressed in #1171. Closing for now
